### DISCUSSION TITLE
I need help. Is my code right? Thank you.

### DISCUSSION THREAD

**Discussion by Xaeon23** [community/community#123553] [Programming Help] [answered]
### Body

//create two classes: an Animal class and a Mamal class. 
// create a cow that accepts a name, type and color and has a sound method that moo's her name, type and color. 
 

class Animal {
    constructor(name, type, color)
    {
        this.name = name;
        this.type = type;  
        this.color = color;
    }

    sound()
    {
        console.log(`Mooooo my name is ${this.name}, and I'm a ${this.type}.
        My color is ${this.color}`);
    }

};

class Mamal extends Animal {
    constructor(name, type, color)
    {
        super(name, type, color)
    }
};

const animal1 = new Mamal("Shawn", "Cow", "White");

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by Pulkit1822** [comment]
**Code Review and Explanation**
===============================

Let's review and discuss the code you provided:

```javascript
// Create two classes: an Animal class and a Mammal class.
// Create a cow that accepts a name, type, and color and has a sound method that moos her name, type, and color.

class Animal {
    constructor(name, type, color) {
        this.name = name;
        this.type = type;
        this.color = color;
    }

    sound() {
        console.log(`Mooooo my name is ${this.name}, and I'm a ${this.type}.
        My color is ${this.color}`);
    }
}

class Mammal extends Animal {
    constructor(name, type, color) {
        super(name, type, color);
    }
}

const animal1 = new Mammal("Shawn", "Cow", "White");
```

### Code Explanation:

1. **Animal Class:**
   - The `Animal` class is defined with a constructor that initializes `name`, `type`, and `color` properties.
   - It has a `sound` method that logs a message with the animal's name, type, and color.

2. **Mammal Class:**
   - The `Mammal` class extends the `Animal` class.
   - It has a constructor that calls the parent class constructor using `super`.

3. **Creating an Instance:**
   - An instance of the `Mammal` class named `animal1` is created with the values "Shawn", "Cow", and "White".

### Code Review:

- The code structure and logic seem correct.
- The classes are defined appropriately with the necessary properties and methods.
- The inheritance relationship between `Animal` and `Mammal` classes is set up correctly.
- The instance creation of `animal1` with the provided values is valid.

### Suggestions:

- Ensure that the `sound` method is called on the `animal1` instance to see the expected output.
- Consider adding error handling or validation logic if needed.
- Test the code with different inputs to verify its functionality.

By following these suggestions, you can further enhance the code's robustness and ensure it meets your requirements.

**Mark as Correct Answer ✅**
-------------------------

If you found this review helpful, please mark me as the correct answer. Happy coding!

**Note:** Remember to test your code thoroughly to ensure it behaves as expected in different scenarios.

**Comment by davevad93** [comment, marked as answer]
Hi @Xaeon23 ,

Yes your code overall is fine, the only thing I would add to it is a call to the `sound()` method on the `animal1` instance, which prints out the sound of the animal along with its name, type, and color:

```JavaScript
animal1.sound();
```
Result:

`Mooooo my name is Shawn, and I'm a Cow. My color is White`

**Reply by ebndev** [reply]
Hey @Xaeon23 - did this code above help? 😄
